protected/humhub/modules/user/models/fieldtype/Template.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace humhub\modules\user\models\fieldtype;
+
+use humhub\helpers\ArrayHelper;
+use humhub\modules\user\models\ProfileField;
+use Twig\Environment;
+use Twig\Loader\ArrayLoader;
+use Yii;
+
+/**
+ * Template is a virtual profile field that combines other profile fields using a Twig template.
+ *
+ * @package humhub\modules\user\models\fieldtype
+ * @since 1.17
+ */
+class Template extends BaseType
+{
+    public $template;
+
+    public $type = 'text';
+
+    public $isVirtual = true;
+
+    public function rules()
+    {
+        return [
+            [['template'], 'string', 'max' => 255],
+        ];
+    }
+
+    public function getUserValue($user, $raw = true): ?string
+    {
+        $variables = ArrayHelper::map(
+            $user->profile->getProfileFields(null, [static::class]),
+            'internal_name',
+            function (ProfileField $profileField) use ($user) {
+                return $profileField->getUserValue($user);
+            },
+        );
+
+        $twig = new Environment(new ArrayLoader([]));
+
+        return $twig->createTemplate($this->template)->render($variables);
+    }
+
+    public function getFormDefinition($definition = [])
+    {
+        return parent::getFormDefinition([
+            get_class($this) => [
+                'type' => 'form',
+                'title' => Yii::t('UserModule.profile', 'Select field options'),
+                'elements' => [
+                    'template' => [
+                        'type' => 'textarea',
+                        'label' => Yii::t('UserModule.profile', 'Template'),
+                        'class' => 'form-control autosize',
+                        'hint' => Yii::t('UserModule.profile', 'Twig template that will be used to render this field. You can use the internal names as variables.'),
```

Comment:
Maybe we can add some exmaple vars like firstname here?

Reply:
@luke- done